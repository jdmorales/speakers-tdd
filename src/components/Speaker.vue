<template>
  
</template>

<script>
export default {
  props: {
    speaker: {
      type: Object,
      required: true
    }
  }
}
</script>

<style>

</style>