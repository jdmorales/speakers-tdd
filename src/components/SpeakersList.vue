<template>
  <div class="columns">
    <div class="column" v-for="speaker in speakers" :key="speaker.username">
      <Speaker :speaker="speaker"/>
    </div>
  </div>
</template>

<script>
import Speaker from '@/components/Speaker'
export default {
  components: {
    Speaker
  },
  props: {
    speakers: {
      type: Array,
      required: true
    }
  }
}
</script>

<style>

</style>