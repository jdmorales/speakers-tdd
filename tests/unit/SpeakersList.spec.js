import { shallowMount } from '@vue/test-utils'
import SpeakersList from '@/components/SpeakersList'
import Speaker from '@/components/Speaker'

describe('SpeakersList', () => {
  let props
  let speaker = {
    username: '@myname',
    name: 'My Name'
  }
  beforeEach(() => {
    // Creating the props to pass when mounting
    props = {
      speakers: [speaker]
    }
  })

  const build = () => {
    const wrapper = shallowMount(SpeakersList, {
      // Add props while mounting
      propsData: props
    })

    return {
      wrapper,
      Speakers: () => wrapper.findAll(Speaker)
    }
  }

  it('it renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it renders the right', () => {
    // arrange
    const { wrapper, Speakers } = build()
    // assert
    expect(Speakers().length).toBe(1)
    wrapper.setProps({
      speakers: [speaker, {
        ...speaker,
        username: '@username'
      }]
    })
    expect(Speakers().length).toBe(2)
  })
})
