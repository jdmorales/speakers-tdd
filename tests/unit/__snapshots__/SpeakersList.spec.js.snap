// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersList renders the component 1`] = `
<div class="columns">
  <div class="column">
    <speaker-stub speaker="[object Object]"></speaker-stub>
  </div>
</div>
`;
